source/common/stats/allocator_impl.cc

```diff
@@ -38,6 +25,23 @@ const char AllocatorImpl::DecrementToZeroSyncPoint[] = "decrement-zero";
 AllocatorImpl::~AllocatorImpl() {
   ASSERT(counters_.empty());
   ASSERT(gauges_.empty());
+
+  // Move deleted stats into the sets for removeFromSetLockHeld to function.
+  for (auto& counter : deleted_counters_) {
+    // Assert that there were no duplicates.
+    ASSERT(counters_.count(counter.get()) == 0);
+    counters_.insert(counter.get());
```

Comment:
You can do this in one statement with `ASSERT(counters.insert(counter.get()).second)`, or in two with
```
auto insertion = counters.insert(counter.get());
ASSERT(insertion.second);
```
The latter feels more like it more closely mirrors the pattern in removeFromSetLockHeld, and is just as fast (only does one map operation).

Reply:
Done

Reply:
Done